python/taichi/linalg/sparse_matrix.py

```diff
@@ -254,10 +255,10 @@ def build(self, dtype=f32, _format='CSR'):
         taichi_arch = get_runtime().prog.config().arch
         if taichi_arch == _ti_core.Arch.x64 or taichi_arch == _ti_core.Arch.arm64:
             sm = self.ptr.build()
-            return SparseMatrix(sm=sm)
+            return SparseMatrix(sm=sm, dtype=dtype)
         if taichi_arch == _ti_core.Arch.cuda:
             sm = self.ptr.build_cuda()
-            return SparseMatrix(sm=sm)
+            return SparseMatrix(sm=sm, dtype=dtype)
```

Comment:
Can we give an error message here for `ti.f64` dtype on CUDA backend?

Reply:
It's probably better to raise the error before doing `sm = self.ptr.build_cuda()`.